### PR TITLE
Update lib - User can choose if calc MD5 from encrypted or decrypted file

### PR DESCRIPTION
**Pull request by zinkett**
## Description of Change
I'm using OTA with pre-encrypted bin file.
So it's easy to calculate the MD5 of the encrypted file before send it to device.
Update function at the moment is calculating the MD5 of the decrypted file.
With this commit with
`Update.setMD5(md5, false);`
the check of MD5 will be done of encrypted file. Otherwise it will be calculate of the decrypted file.

## Tests scenarios
I have tested my Pull Request on Arduino-esp32 core v3.0.6 with ESP32

### CODE REVIEW

**Comment by SuGlider** [comment]
> [CLA assistant check](https://cla-assistant.io/espressif/arduino-esp32?pullRequest=10510) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/espressif/arduino-esp32?pullRequest=10510) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/arduino-esp32?pullRequest=10510) it.

@zinkett - Please sign electronically the CLA using your Github username, if you agree to it.

**Comment by SuGlider** [comment]
@zinkett - There are some compilation error within this PR (pointed by CI). Please check it at https://github.com/espressif/arduino-esp32/actions/runs/11479892688/job/31957134561?pr=10510

File: libraries/Update/src/Update.h [hunk 1]
```diff
@@ -137,8 +137,9 @@ class UpdateClass {
 
   /*
       sets the expected MD5 for the firmware (hexString)
+      calc_post_decryption if true che update library calc MD5 after decryption, if false the calc is before decryption
     */
-  bool setMD5(const char *expected_md5);
+  bool setMD5(const char *expected_md5, bool calc_post_decryption);
```
Comment (lucasssvaz) [inline libraries/Update/src/Update.h:142]:
```suggestion
  bool setMD5(const char *expected_md5, bool calc_post_decryption=true);
```

File: libraries/Update/src/Update.h [hunk 2]
```diff
@@ -137,8 +137,9 @@ class UpdateClass {
 
   /*
       sets the expected MD5 for the firmware (hexString)
+      calc_post_decryption if true che update library calc MD5 after decryption, if false the calc is before decryption
```
Comment (lucasssvaz) [inline libraries/Update/src/Update.h:140]:
```suggestion
      if calc_post_decryption is true, the update library will calculate the MD5 after the decryption, if false the calculation occurs before the decryption
```

File: libraries/Update/src/Updater.cpp [hunk 3]
```diff
@@ -5,7 +5,7 @@
  */
 
 #include "Update.h"
-#include "Arduino.h"
+#include "Arduino.h"
```
Comment (lucasssvaz) [inline libraries/Update/src/Updater.cpp:8]:
```suggestion
#include "Arduino.h"
```

File: libraries/Update/src/Updater.cpp [hunk 4]
```diff
@@ -446,12 +453,13 @@ bool UpdateClass::_verifyEnd() {
   return false;
 }
 
-bool UpdateClass::setMD5(const char *expected_md5) {
+bool UpdateClass::setMD5(const char *expected_md5, bool calc_post_decryption=true) {
```
Comment (lucasssvaz) [inline libraries/Update/src/Updater.cpp:456]:
```suggestion
bool UpdateClass::setMD5(const char *expected_md5, bool calc_post_decryption) {
```

**Review by zinkett — COMMENTED** [review]
thank you @lucasssvaz

**Review by lucasssvaz — APPROVED** [review]
LGTM

**Review by me-no-dev — APPROVED** [review]

**Pull request merged**